src/api/interfaces/subscription.rs

```diff
@@ -0,0 +1,104 @@
+/*
+   Copyright 2019 Supercomputing Systems AG
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+	   http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+use crate::{
+	api::{error::Error, Api, ApiResult},
+	rpc::{HandleSubscription, Subscribe},
+};
+pub use ac_node_api::{events::EventDetails, StaticEvent};
+use ac_node_api::{DispatchError, Events};
+use ac_primitives::{ExtrinsicParams, FrameSystemConfig};
+use log::*;
+use serde::de::DeserializeOwned;
+use sp_core::storage::StorageChangeSet;
+
+// FIXME: This should rather be implemented directly on the
+// Subscription return value, rather than the api. Or directly
```

Comment:
I don't like how it's currently done. But changing this would break the interface. So that should be done in a different PR

Reply:
Now, we only have events here. Maybe we can call it `EventSubscription`?

Reply:
Ah, actually, following the rust convention, that should rather be a verb noun. How about `SubscribeEvents`?

Reply:
Yes, I agree that is even better, but the last thing would to also rename the file accordingly to `subscribe_events` :)

Reply:
Done :)